core/dbt/parser/manifest.py

```diff
@@ -599,6 +652,173 @@ def write_manifest_for_partial_parse(self):
         except Exception:
             raise
 
+    def write_artifacts(self):
+        # The manifest.json is written out in a task, so we're not writing it here
```

Comment:
What do you think about: 
1. renaming this method to `write_publication` (accepting a manifest + target_path, similar to write_path) and calling it alongside `write_manifest` in requires.py
2. Bundling the two calls (write_manifest, write_publication) together in a `write_artifact` method which is  called in requires.py. 

Given we're going to be introducing a semantic graph artifact soon, it'd be good to set a pattern for writing manifest-like objects consistently. Especially if there isn't a use case where we'd want to write one of the manifest, publication, or semantic graph separately from one another.

Reply:
I separated this out into a function in core/dbt/parser/manifest.py, but didn't change to calling it in requires.py because it would get re-created on every single command invocation, which feels like too much to me. Right now it's recreated if parsing happens or the publication artifacts changed. Since it's now a function it can be easily moved somewhere else if necessary.